packages/perseus/src/widgets/interactive-graphs/graphs/polygon.tsx

```diff
@@ -31,35 +33,52 @@ export function renderPolygonGraph(
 }
 
 type Props = MafsGraphProps<PolygonGraphState>;
+type StatefulProps = MafsGraphProps<PolygonGraphState> & {
+    graphConfig: GraphConfig;
+    polygonRef: React.RefObject<SVGPolygonElement>;
+    pointsRef: React.MutableRefObject<(SVGElement | null)[]>;
+    lastMoveTimeRef: React.MutableRefObject<number>;
+    left: number;
+    top: number;
+    dragging: boolean;
+    points: Coord[];
+    constrain: (p: any) => any;
+    hovered: boolean;
+    setHovered: React.Dispatch<React.SetStateAction<boolean>>;
+    focusVisible: boolean;
+    setFocusVisible: React.Dispatch<React.SetStateAction<boolean>>;
+};
 
-const LimitedPolygonGraph = (props: Props) => {
-    const [hovered, setHovered] = React.useState(false);
-    // This is more so required for the re-rendering that occurs when state
-    // updates; specifically with regard to line weighting and polygon focus.
-    const [focusVisible, setFocusVisible] = React.useState(false);
-
+const PolygonGraph = (props: Props) => {
     const {dispatch} = props;
+    const {numSides, coords, snapStep, snapTo = "grid"} = props.graphState;
+    const graphConfig = useGraphConfig();
+
+    // Ref to implement the dragging behavior on a Limited/Closed Polygon.
+    const polygonRef = React.useRef<SVGPolygonElement>(null);
+    // Ref to manage dynamic focus for the points in Unlimited Polygon
+    const pointsRef = React.useRef<Array<SVGElement | null>>([]);
+    // Ref to manage the last move tim for a Limited Polygon.
+    const lastMoveTimeRef = React.useRef<number>(0);
```

Comment:
Interesting usage! I don't think I've seen a ref used for anything other than an HTML element before, but I see what you're doing here.

Reply:
Me neither 🤣

Reply:
Alternatively, I suppose you could make this a state value, and you can pass in the value and its setter into the limited and unlimited polygon graphs. But that seems more convoluted.